lightspeed-stack/Containerfile.lsc

```diff
@@ -0,0 +1,75 @@
+# ======================================================
+# Transient image to construct Python venv
+# ------------------------------------------------------
+FROM registry.access.redhat.com/ubi9/ubi-minimal AS builder
+
+ARG APP_ROOT=/app-root
+RUN microdnf install -y --nodocs --setopt=keepcache=0 --setopt=tsflags=nodocs \
+    python3.11 python3.11-devel python3.11-pip
+WORKDIR /app-root
+
+# UV_PYTHON_DOWNLOADS=0 : Disable Python interpreter downloads and use the system interpreter.
+ENV UV_COMPILE_BYTECODE=0 \
+    UV_LINK_MODE=copy \
+    UV_PYTHON_DOWNLOADS=0
+
+# Install uv package manager
+RUN pip3.11 install uv
+
+# Add explicit files and directories
+# (avoid accidental inclusion of local directories or env files or credentials)
+COPY lightspeed-stack/pyproject.toml lightspeed-stack/uv.lock LICENSE.md README.md ./
+
+RUN uv sync --no-install-project --no-dev
+# ======================================================
+
+# ======================================================
+# Final image without uv package manager
+# ------------------------------------------------------
+FROM quay.io/lightspeed-core/lightspeed-stack:latest
+
+USER 0
+
+ARG APP_ROOT=/app-root
+RUN microdnf install -y --nodocs --setopt=keepcache=0 --setopt=tsflags=nodocs python3.11
+WORKDIR /app-root
+
+# PYTHONDONTWRITEBYTECODE 1 : disable the generation of .pyc
+# PYTHONUNBUFFERED 1 : force the stdout and stderr streams to be unbuffered
+# PYTHONCOERCECLOCALE 0, PYTHONUTF8 1 : skip legacy locales and use UTF-8 mode
+ENV PYTHONDONTWRITEBYTECODE=1 \
+    PYTHONUNBUFFERED=1 \
+    PYTHONCOERCECLOCALE=0 \
+    PYTHONUTF8=1 \
+    PYTHONIOENCODING=UTF-8 \
+    LANG=en_US.UTF-8
+
+COPY --from=builder --chown=1001:1001 /app-root /app-root
+
+# this directory is checked by ecosystem-cert-preflight-checks task in Konflux
+COPY --from=builder /app-root/LICENSE.md /licenses/
+
+# Add executables from .venv to system PATH
+ENV PATH="/app-root/.venv/bin:$PATH"
+
+ENV LLAMA_STACK_CONFIG_DIR=/.llama/data
+
+# Data and configuration
+RUN mkdir -p /.llama/distributions/ansible-chatbot
+RUN mkdir -p /.llama/data/distributions/ansible-chatbot
+ADD ansible-chatbot-run.yaml /.llama/distributions/ansible-chatbot
+ADD providers.d /.llama/providers.d
```

Comment:
there is a problem here... the `providers.d/` was previously generated by the `llama stack build` command... now that we don't run it, what? :) should we update the `setup` target for `make`?

Reply:
yeah I mean it is supposed the directory doesn't exist, if you run `make clean` , it is deleted, so the build fails

Reply:
I am not sure what llama stack does during the build to generate that folder, but I guess it is just a copy paste of the `~/.llama/providers.d` one

Reply:
so maybe just using `~/.llama/providers.d` instad makes it work

Reply:
Fixed. New commit pushed. 

I copy `providers.d` into `<repository-root>/llama-stack/providers.d` now

Keeping all _transient_ build folders in the repository root is consistent with our RAG setup.